### PR TITLE
[SPARK-29663][SQL] Support sum with interval type values

### PR DESCRIPTION
**Pull request by yaooqinn**
### What changes were proposed in this pull request?

sum support interval values

### Why are the changes needed?

Part of SPARK-27764 Feature Parity between PostgreSQL and Spark

### Does this PR introduce any user-facing change?

yes, sum can evaluate intervals
### How was this patch tested?

add ut

### CODE REVIEW

**Comment by yaooqinn** [comment]
cc @cloud-fan @MaxGekk @dongjoon-hyun @wangyum

**Review by MaxGekk — COMMENTED** [review]
Could you add an example with intervals to https://github.com/apache/spark/pull/26325/files#diff-84a56c8f7c77992f9f7294cea56734ceL30-L36

**Comment by yaooqinn** [comment]
@MaxGekk good advice, thanks.

**Comment by yaooqinn** [comment]
retest this please

**Comment by yaooqinn** [comment]
@MaxGekk please recheck thanks

**Review by MaxGekk — APPROVED** [review]

**Comment by yaooqinn** [comment]
@MaxGekk Can we have this merged？ Or you may help cc some other reviewer. thank youvery much.

**Comment by MaxGekk** [comment]
@yaooqinn I don't have permissions to merge this. You need someone from the list https://spark.apache.org/committers.html

**Comment by yaooqinn** [comment]
cc @maropu @HyukjinKwon thanks

File: sql/core/src/test/resources/sql-tests/inputs/sum.sql [hunk 1]
```diff
@@ -0,0 +1,32 @@
+-- sum
```
Comment (cloud-fan) [inline sql/core/src/test/resources/sql-tests/inputs/sum.sql]: shall we make it clear that this only test interval sum?

Comment (cloud-fan) [reply sql/core/src/test/resources/sql-tests/inputs/sum.sql]: or you can add tests to `group-by.sql`

Comment (yaooqinn) [reply sql/core/src/test/resources/sql-tests/inputs/sum.sql]: I can move them to `group-by.sql`

File: sql/core/src/test/resources/sql-tests/inputs/group-by.sql [hunk 2]
```diff
@@ -162,3 +162,34 @@ SELECT count(*) FROM test_agg WHERE count(*) > 1L;
...
 SELECT count(*) FROM test_agg WHERE k = 1 or k = 2 or count(*) + 1L > 1L or max(k) > 1;
 
+-- sum interval values
+-- null
+select sum(cast(v as interval)) from VALUES ('1 seconds'), ('2 seconds'), (null) t(v) where v is null;
+
+-- empty set
+select sum(cast(v as interval)) from VALUES ('1 seconds'), ('2 seconds'), (null) t(v) where 1=0;
+
+--
```
Comment (cloud-fan) [inline sql/core/src/test/resources/sql-tests/inputs/group-by.sql]: `basic interval sum`

File: sql/core/src/test/resources/sql-tests/inputs/group-by.sql [hunk 3]
```diff
@@ -162,3 +162,34 @@ SELECT count(*) FROM test_agg WHERE count(*) > 1L;
...
+select sum(cast(v as interval)) from VALUES ('1 seconds'), ('2 seconds'), (null) t(v);
+select sum(cast(v as interval)) from VALUES ('-1 seconds'), ('2 seconds'), (null) t(v);
+select sum(cast(v as interval)) from VALUES ('-1 seconds'), ('-2 seconds'), (null) t(v);
+select sum(cast(v as interval)) from VALUES ('-1 weeks'), ('2 seconds'), (null) t(v);
+
+--group by
```
Comment (cloud-fan) [inline sql/core/src/test/resources/sql-tests/inputs/group-by.sql]: one space after `--`

File: sql/core/src/test/resources/sql-tests/inputs/group-by.sql [hunk 4]
```diff
@@ -162,3 +162,34 @@ SELECT count(*) FROM test_agg WHERE count(*) > 1L;
...
+select
+    i,
+    sum(cast(v as interval))
+from VALUES (1, '-1 weeks'), (2, '2 seconds'), (3, null), (1, '5 days') t(i, v)
+group by i;
+
+--having
```
Comment (cloud-fan) [inline sql/core/src/test/resources/sql-tests/inputs/group-by.sql]: ditto

Comment (yaooqinn) [reply sql/core/src/test/resources/sql-tests/inputs/group-by.sql]: updated with all above comments, thank for reviewing

**Review by maropu — APPROVED** [review]

**Comment by cloud-fan** [comment]
thanks, merging to master!

**Pull request closed without merging**